aws/resource_aws_codeartifact_repository.go

```diff
@@ -116,6 +117,21 @@ func resourceAwsCodeArtifactRepositoryCreate(d *schema.ResourceData, meta interf
 	repo := res.Repository
 	d.SetId(aws.StringValue(repo.Arn))
 
+	if v, ok := d.GetOk("external_connections"); ok {
```

Comment:
Need to handle multiple external connections?

Reply:
At the moment on connection per repo is supported per docs but this may future proof to treat it as a set. Will change.

Reply:
If you set `MaxItems: 1,` then one of the linters will complain that it should be `TypeList`. It seem like if the API enforces max of 1 connection then we should have that in the schema and stick with `TypeList`.